crates/ty_python_semantic/resources/mdtest/generics/legacy/classes.md

```diff
@@ -354,6 +354,8 @@ class C(Generic[T]):
     @overload
     def __init__(self: "C[bytes]", x: bytes) -> None: ...
     @overload
+    def __init__(self: "C[int]", x: bytes) -> None: ...
```

Comment:
Maybe not worth a separate follow-up PR, but wouldn't we ideally have a case where the second argument is optional and so we can have multiple matching overloads, and show that we pick the first?

Reply:
This gives us two matching overloads if you provide a `bytes` parameter, and we show that we pick the first (giving `C[bytes]` instead of `C[int]` or `C[bytes | int]`).  Maybe deserves a comment, since the argument / specialization pattern across the overloads is subtle?